utils/configuration.go

```
@@ -0,0 +1,94 @@
+// cannot distribute API keys as part of our open source application. This necessitates every sctl admin
+// to create a GCP project, and issue a client credential for authenticating users.
+// We will need these values throughout the lifetime of credential issuance so we'll serialize these
+// not secret, but EULA bound credentials in the persistent configuration.
+
+// A documented model of the proposed configuration is as follows
+// Note the prefix of GCP on the cloud options. The intent is to namespace
+// so in the future adding additional clouds will not be disruptive to the structure of sctl.
+//
+//
+// {
+//    // client is responsible for recording the base64 encoded application client credentials.
+// these API keys only identify the requesting application, and broker the authentication
+// between operator and GCP
+//     "gcp_client_config": {
+//         "data": "BASE64 Data",
+//     }
+// }
+
+// Configuration ...
+type Configuration struct {
+	GoogleClient Client `json:"gcp_client_config,omitempty"`
+}
+
+// Client is a sub-object warehousing the client configuration data
+// provided by google. https://console.developers.google.com/
+type Client struct {
+	Data string `json:"data", omitempty`
+}
+
+// Save ...
+func (c *Configuration) Save() error {
+	jsonData, _ := json.MarshalIndent(&c, "", " ")
+	ioutil.WriteFile(c.configPath(), jsonData, os.FileMode(0600))
+
+	return nil
+}
+
+// Load ...
+func (c *Configuration) Load() error {
+	f, err := ioutil.ReadFile(c.configPath())
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal([]byte(f), &c)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Init will generate the configuration path for sctl in the system UserConfigDir
+func (c *Configuration) Init() {
+	os.MkdirAll(c.configDir(), 0775)
+}
+
+func (c *Configuration) configDir() string {
+	userConfig, err := os.UserConfigDir()
+
+	if err != nil {
+		log.Fatalf("Unable to determine Operating System UserConfigDir() - %s", err)
```

should return error instead of log.fatal